src/otx/api/usecases/exportable_code/prediction_to_annotation_converter.py

```diff
@@ -342,15 +341,14 @@ def convert_to_annotation(self, predictions: np.ndarray, metadata: Dict[str, Any
         Returns:
             AnnotationSceneEntity: OTX annotation scene entity object.
         """
-        pred_label = predictions >= metadata.get("threshold", 0.5)
-
-        label = self.anomalous_label if pred_label else self.normal_label
-        probability = (1 - predictions) if predictions < 0.5 else predictions
+        assert predictions.pred_score is not None
+        assert predictions.pred_label is not None
+        label = self.anomalous_label if predictions.pred_label == "Anomaly" else self.normal_label
 
         annotations = [
             Annotation(
                 Rectangle.generate_full_box(),
-                labels=[ScoredLabel(label=label, probability=float(probability))],
+                labels=[ScoredLabel(label=label, probability=predictions.pred_score)],
```

Comment:
are we 100% sure that `predictions.pred_score` is always float?

Reply:
I guess https://github.com/openvinotoolkit/model_api/blob/master/model_api/python/openvino/model_api/models/anomaly.py#L95. Not sure if np.item() returns ints

Reply:
Would it be a safe bet if we also add an assert statement for this one ?

Reply:
Done!